compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentHandlerGeneratorTest.kt

```diff
@@ -1547,6 +1558,76 @@ class ContributesSubcomponentHandlerGeneratorTest {
     }
   }
 
+  @Test fun `an @ContributeSubcomponent class can be generated`() {
+    val codeGenerator = object : CodeGenerator {
```

Comment:
Custom code generator in unit tests!

Reply:
Between this (which forces Psi resolution) and passing in the previous compilation result (which forces Descriptor resolution), there's nothing we can't _easily_ test.  I love it!